app_dart/lib/src/request_handlers/update_task_status.dart

```diff
@@ -95,44 +96,30 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
       task.status = newStatus;
       task.endTimestamp = DateTime.now().millisecondsSinceEpoch;
     }
-
-    await runTransactionWithRetries(() async {
-      await datastore.db.withTransaction<void>((Transaction transaction) async {
-        transaction.queueMutations(inserts: <Task>[task]);
-        await transaction.commit();
-      });
-    });
-
+    await datastore.insert(<Task>[task]);
     if (task.endTimestamp > 0) {
       await _insertBigquery(commit, task);
     }
 
     // TODO(tvolkert): PushBuildStatusToGithub
-
-    if (newStatus == Task.statusSucceeded && scoreKeys.isNotEmpty) {
-      for (String scoreKey in scoreKeys) {
-        await runTransactionWithRetries(() async {
-          await datastore.db
-              .withTransaction<void>((Transaction transaction) async {
-            final TimeSeries series =
-                await _getOrCreateTimeSeries(transaction, task, scoreKey);
-            final num value = resultData[scoreKey] as num;
-
-            final TimeSeriesValue seriesValue = TimeSeriesValue(
-              key: series.key.append(TimeSeriesValue),
-              createTimestamp: DateTime.now().millisecondsSinceEpoch,
-              revision: commit.sha,
-              taskKey: task.key,
-              value: value.toDouble(),
-            );
-
-            transaction.queueMutations(inserts: <TimeSeriesValue>[seriesValue]);
-            await transaction.commit();
-          });
-        });
-      }
+    if (newStatus != Task.statusSucceeded || scoreKeys.isEmpty) {
+      return UpdateTaskStatusResponse(task);
```

Comment:
When does this occur? At a glance it's hard to tell why we exit early here (other than to maintain the existing logic), and it would be good to document it.

Reply:
Actually I just rewrote the logic using the datastore object but looking this part closely seems like the intent was an optimization but in reality it is duplicating code. If scoreKeys is empty the code inside the loop does not execute and goes directly to UpdateTaskStatusResponse.

Removed those lines.